ethnode/rpc.go

```diff
@@ -296,3 +312,28 @@ func RemoteNode(client *rpc.Client) (EthNode, error) {
 
 	return node, nil
 }
+
+// EnodeEqual returns whether two enode:// strings are equivalent after
+// normalization, comparing only the enodeID and remote address.
+// Failure to parse returns false. Remote address "[::]" acts as a wildcard.
+func EnodeEqual(a, b string) bool {
+	uriA, err := url.Parse(a)
+	if err != nil || uriA.User == nil {
+		return false
+	}
+	uriB, err := url.Parse(b)
+	if err != nil || uriB.User == nil {
+		return false
+	}
+
+	if uriA.User.Username() != uriB.User.Username() {
+		return false
+	}
+	if uriA.Hostname() == "::" || uriB.Hostname() == "::" {
+		if uriA.Port() != uriB.Port() {
```

Comment:
Are we sure we want to equate on port numbers?  Won't the ports _always_ mismatch if we are comparing remote- and locally- initiated connections?  That is, if I connect to nodeB, the remote address will be `:30303` but if they connect to me the remote address will be an ephemeral port number.

Also, since this uses url.Parse like I recommended for enodeToId maybe we should have a `NewEnodeURI(uri string) (*EnodeURI, error)` method that returns a `EnodeURI` struct with fields like `ID`, `Address`, and `Port`, and possibly helpers like `IsWildcard()`?

Reply:
(Oops misclicked on resolve)

> Are we sure we want to equate on port numbers? 

Good question! The way it's being used right now, it's only comparing the reported remote addresses from the pool vs local peers' remote addresses, so it's always remote vs remote (external ports).

The reason I wanted to equate port numbers at all is for the scenario if a node goes down, then comes back up on a different port for some reason, that should force an invalidation -> disconnect -> eventual reconnect next time it's returned from the pool.

But you're right, it's risky making this a public helper because it's a foot-gun if people aren't careful about that. At minimum, I'll move it to be a private helper where it's being used.

> Also, since this uses url.Parse like I recommended for enodeToId maybe we should have a `NewEnodeURI(uri string) (*EnodeURI, error)` method that returns a `EnodeURI` struct with fields like `ID`, `Address`, and `Port`, and possibly helpers like `IsWildcard()`?

I like this idea. Not sure if I'll fit it into this PR, but definitely :+1:.

Reply:
Removed it altogether for now and opened https://github.com/vipnode/vipnode/issues/76